use one(::SymmetricGroup) instead of (G::SymmetricGroup)()

kalmarek opened the pull request:
to follow group interface of oscar, e.g.
https://github.com/oscar-system/Oscar.jl/blob/ae0a3b0e4daa7fdeff73241b80bbeadc39cfe298/src/Groups/GAPGroups.jl#L189

(@fingolfin, is there a doc with the interface we drafted?)

btw. given that GAP permutations are the permutations of oscar I'd also stop exporting `perm` (why lowercase?) to allow this
https://github.com/oscar-system/Oscar.jl/blob/ae0a3b0e4daa7fdeff73241b80bbeadc39cfe298/src/Groups/GAPGroups.jl#L258
to move on.

Also: given the remarkably small impact of this change, is it really worth keeping here? (all usages inside AA only need the storage vector of a permutation).

@rfourquet @fingolfin @thofma

EDIT: In the meantime I forgot why have I started this:
* added compact printing of perms :)
```julia
julia> p = randperm(100);

julia> Perm(p)
(1,18,54,50,69,80,58,47,81,24,70,16,93,46,31,90,56,5,57,51,99,76,49,3,62,40,10,27,78,14,25,82,53,45,37,13,44,22,87,97,21,3 … 

```

thofma commented:
This just changed in the invocation in the test/examples, right?

rfourquet commented on src/generic/PermGroups.jl:
```diff
@@ -340,15 +340,34 @@ function setpermstyle(format::Symbol)
+   end
+end
+
+function _permstring(p::Perm, width::Integer)
```
This method could probably be made eventually to accept an `io` argument and writing to it (via the `join(io, ...)` method), instead of creating a big string. But who cares, until we get complaints :)

kalmarek commented:
> This just changed in the invocation in the test/examples, right?

@thofma I'm not sure what do you mean. I changed `G()` → `one(G)` everywhere, that's it.

I added deprecations, so that this isn't braking;
@rfourquet I took your advice and renamed the function to `_print_perm` in resemblance to `Base.print_array`, since this function actually prints something now ;) You can always get the string through `sprint(show, p)`.

thofma commented:
I mean, we did not remove the method for `G()`, right? Just invocations were changed and not the actual method removed.

rfourquet commented on src/generic/PermGroups.jl:
```diff
@@ -365,9 +365,9 @@ function _permstring(p::Perm, width::Integer)
       end
       out = "("*join(res, ")(")*")"
```
Here the full string is still created despite the `io` argument elsewhere. Avoiding this would require writing each element of `res` to `io` after checking that the overall length doesn't overflow; but this is probably not a worthy complication.

kalmarek replied:
Ok, that's what I did ;) One could even avoid allocating the cycle string, but I'm not that compulsive at this moment ;P

fingolfin commented:
The draft groups interface is described here: https://hackmd.io/HrSzCeq_TpyLeG6PLFAbDg -- really need to turn this into a proper text, flesh it out, and also set up a conformance test to be used in the Oscar groups code...

kalmarek commented:
@thofma Group interface we forged does not require `G()`, the agreed way of getting the identity is `one(G)`;
at this moment I added `G()` to deprecated functions, to allow a non-breaking release; If the next major version is released this deprecation should be removed.

thofma commented:
Thanks for the explanation. I overlooked the changes in Deprecation.jl.

fingolfin commented:
Somebody who has the rights, please merge :-)

thofma commented on src/Deprecations.jl:
```diff
@@ -19,3 +19,8 @@
 
 @deprecate MatrixAlgebra(R::Ring, n::Int, cached::Bool) MatrixAlgebra(R, n, cached = cached)
+
+function (G::Generic.SymmetricGroup)()
```
Why not @deprecate? This version does not print the backtrace, so I don't know where it is called :(

kalmarek replied:
no deprecations are showed in and after julia-1.5; use `julia --depwarn=yes` to see them;

moreover `@deprecate` macro does not parse `(G::Generic.SymmetricGroup)()`;

thofma replied:
I am running it with `depwarn=yes`:
```julia
julia> SymmetricGroup(3)()                                                                                
┌ Warning: (::SymmetricGroup)() to get the group identity is deprecated, use one(::SymmetricGroup) instead
│   caller = ip:0x0                                                                                       
└ @ Core :-1                                                                                              
()                                                                                                        
```
No backtrace.

thofma replied:
I am not talking about seeing it or not, but about the backtrace.

thofma replied:
Annoying that the macro does not work :-(
